smac/optimizer/acquisition.py

```diff
@@ -157,14 +157,20 @@ def _compute(self, X: np.ndarray, **kwargs):
                              'eta=<int>) to inform the acquisition function '
                              'about the current best value.')
 
-        z = (self.eta - m - self.par) / s
-        f = (self.eta - m - self.par) * norm.cdf(z) + s * norm.pdf(z)
+        def calculate_f():
+            z = (self.eta - m - self.par) / s
+            return (self.eta - m - self.par) * norm.cdf(z) + s * norm.pdf(z)
+
         if np.any(s == 0.0):
             # if std is zero, we have observed x on all instances
             # using a RF, std should be never exactly 0.0
-            self.logger.warn("Predicted std is 0.0 for at least one sample.")
-            f[s == 0.0] = 0.0
-
+            self.logger.warning("Predicted std is 0.0 for at least one sample.")
```

Comment:
While this code would certainly work it looks overly complicated. How about suppressing the warning and replacing the returned value by an acquisition value of 0. This should make the code easier to understand.

Reply:
The warning was a ZeroDivision Error so this is a major issue that should not be suppressed in my opinion. The code now has the semantics that -- I guess -- were initially intended, but without the warning. Also, how a zero division error is handled by the CPU/FPU, OS and python runtime environment is specific to implementation or configuration so I want to avoid this Exception to not add potential overhead by Exception handling or other routines running. From my engineering point of view I consider it bad practice to program with Exceptions when they can be avoided. I added a comment telling what the if-branch is doing.

Reply:
It should be specific to numpy and its configuration, right? Anyway, it looks fine to me.